src/service/performance-impl.js

```diff
@@ -506,17 +554,75 @@ export class Performance {
         }
       }
 
-      this.tickLayoutShiftScore_();
+      this.tickCumulativeLayoutShiftScore_();
     }
     if (this.supportsLargestContentfulPaint_) {
       this.tickLargestContentfulPaint_();
     }
   }
 
+  /**
+   * Tick the layout shift metric, following the latest CWV standard. This uses
+   * a 5s maximum window with a 1s gap between events to define a "session".
+   * We report the maximum session value over the lifetime of the page as the CLS.
+   *
+   * @param {!LayoutShift} entry
+   */
+  tickNominalLayoutShiftScore_(entry) {
+    const entries = this.layoutShiftsNominal_;
+    if (entries.length > 0) {
+      const first = entries[0];
+      const last = entries[entries.length - 1];
+      if (
+        entry.startTime - last.startTime < CLS_SESSION_GAP &&
+        entry.startTime - first.startTime < CLS_SESSION_MAX
+      ) {
+        // This entry continues the current CLS window.
+        entries.push(entry);
+        return;
+      }
+      // This entry is the start of a new CLS window, but we haven't flushed the old value yet.
+      this.flushNominalLayoutShiftScore_();
+    }
+    entries.push(entry);
+    // Ensure we report the CLS when the session closes. We're not guaranteed
+    // to get more LayoutShift entires, so we need some setTimeout magic to
+    // ensure it happens.
+    this.debouncedFlushNominalLayoutShiftScore_();
+  }
+
+  /**
+   * Records the largest Nominal CLS score, following the latest CWV standard.
+   * See https://web.dev/evolving-cls/
+   */
+  flushNominalLayoutShiftScore_() {
+    const entries = this.layoutShiftsNominal_;
+    const old = this.metrics_.get(TickLabel.CUMULATIVE_LAYOUT_SHIFT);
+    let union = 0;
+    let sum = 0;
+    for (const entry of entries) {
+      if (entry.sources) {
+        for (const source of entry.sources) {
+          union |= getElementType(source.node);
+        }
+      }
+      sum += entry.value;
+    }
+    entries.length = 0;
+    if (old == null || sum > old) {
+      // We'll record the largest windowed CLS.
+      this.metrics_.reset(TickLabel.CUMULATIVE_LAYOUT_SHIFT);
+      this.metrics_.reset(TickLabel.CUMULATIVE_LAYOUT_SHIFT_TYPE_UNION);
+      this.tickDelta(TickLabel.CUMULATIVE_LAYOUT_SHIFT, sum);
```

Comment:
Should calling `tick` on an already reported metric automatically reset it? I.e. so you don't need to do so here as well.